### PR TITLE
New Website

### PR DESCRIPTION
**Pull request by trisyoungs**
This PR removes the old `docs/` documentation served via github-pages and replaces it with `web/` containing a complete Hugo site for Dissolve. Pipelines have been added to automate the build and deployment (via rsync over ssh) to [https://www.projectdissolve.com](https://www.projectdissolve.com).

Notes:
1. "Edit this page on GitHub" links are rewritten manually before deployment, as Hugo inserts a language slug (i.e. `en`) into the url, which is not present in the actual url.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[submodule "dissolve-web/themes/docsy"]
```
Comment (DanNixon) [inline .gitmodules:1]: Why are there two entries here?

File: .azure-pipelines/templates/build-web.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+parameters:
+  - name: hugo_version
+    default: '0.74.3'
+  - name: use_current_hash
+    default: false
+  - name: repo_dir_url
+    default: 'github.com/projectdissolve/dissolve/edit/develop/web'
```
[the remark this answers is not in the document]
Comment (trisyoungs) [reply .azure-pipelines/templates/build-web.yml:7]: It's the root of the web content dir in the repo, and is used on line 67 when rewriting the edit links (which wrongly have the language slug in them).  Used a parameter to try and keep things neater in the actual script.

File: .gitmodules (hunk 1 above)
Comment (trisyoungs) [reply .gitmodules:1]: Bad merge when moving over Simona's PR. Now fixed.

File: .azure-pipelines/templates/build-web.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanNixon) [reply .azure-pipelines/templates/build-web.yml:7]: But this is a dead link, is it modified in some way? Ether way the variable name is misleading.

[the remark this answers is not in the document]
Comment (trisyoungs) [reply .azure-pipelines/templates/build-web.yml:7]: It's only the prefix for the bit that gets changed - admittedly the sed command could just look for `content/en/docs` but I felt that explicitly searching for the whole url was probably safer.  Fair point regarding the name - now changed to be more appropriate.

File: .azure-pipelines/pr.yml [hunk 3]
```diff
@@ -8,6 +8,11 @@ pr:
   branches:
     include:
       - '*'
+  paths:
```
[the remark this answers is not in the document]
Comment (trisyoungs) [reply .azure-pipelines/pr.yml:11]: Is there any doc you can point me to for that?  Had a look around and can't see any way of using dir/file triggers for individual stages from the DevOps docs.

[the remark this answers is not in the document]
Comment (DanNixon) [reply .azure-pipelines/pr.yml:11]: These two go over conditions for stages and jobs:
- https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml
- https://docs.microsoft.com/en-us/azure/devops/pipelines/process/stages?view=azure-devops&tabs=yaml

The syntax is not that similar so may be more effort than it is worth. In this case we should just always build everything.

[the remark this answers is not in the document]
Comment (trisyoungs) [reply .azure-pipelines/pr.yml:11]: And it's not immediately clear how to base conditionals only on specific paths changing, which is required here.  I think leave it for now, and take the original approach.

File: .azure-pipelines/templates/build-web.yml [hunk 4]
```diff
@@ -65,15 +68,18 @@ steps:
...
       secureFile: id_rsa_dd
+    condition: eq('${{ parameters.deploy }}', true)
+    displayName: 'Download Secure Files'
   - task: DownloadSecureFile@1
     name: machineId
-    displayName: 'Download Secure Files'
     inputs:
       secureFile: pd_hostkey
+    condition: eq('${{ parameters.deploy }}', true)
+    displayName: 'Download Secure Files'
```
Comment (DanNixon) [inline .azure-pipelines/templates/build-web.yml:80]: Can we have more useful names, this may as well not be named as it is.

Comment (trisyoungs) [reply .azure-pipelines/templates/build-web.yml:80]: Changed from `deploy` to `push_to_server`.

Comment (DanNixon) [reply .azure-pipelines/templates/build-web.yml:80]: I was referring to `displayName`. `deploy` is fine for a parameter name.

Comment (trisyoungs) [reply .azure-pipelines/templates/build-web.yml]: Reverted, and changed.

**Pull request merged**